omod/src/main/java/org/openmrs/module/metadatamapping/web/rest/MetadataTermMappingResource.java

```diff
@@ -0,0 +1,252 @@
+package org.openmrs.module.metadatamapping.web.rest;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.commons.beanutils.PropertyUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.metadatamapping.MetadataSource;
+import org.openmrs.module.metadatamapping.MetadataTermMapping;
+import org.openmrs.module.metadatamapping.MetadataTermMapping.MetadataReference;
+import org.openmrs.module.metadatamapping.api.MetadataMappingService;
+import org.openmrs.module.metadatamapping.api.MetadataTermMappingSearchCriteriaBuilder;
+import org.openmrs.module.metadatamapping.web.controller.MetadataMappingRestController;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.web.RequestContext;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.annotation.PropertyGetter;
+import org.openmrs.module.webservices.rest.web.annotation.PropertySetter;
+import org.openmrs.module.webservices.rest.web.annotation.Resource;
+import org.openmrs.module.webservices.rest.web.representation.DefaultRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.FullRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.api.PageableResult;
+import org.openmrs.module.webservices.rest.web.resource.impl.AlreadyPaged;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.resource.impl.MetadataDelegatingCrudResource;
+import org.openmrs.module.webservices.rest.web.response.ObjectNotFoundException;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+
+@Resource(name = RestConstants.VERSION_1 + MetadataMappingRestController.METADATA_MAPPING_REST_NAMESPACE
+        + "/metadatatermmapping", supportedClass = MetadataTermMapping.class, supportedOpenmrsVersions = { "1.9.*",
+        "1.10.*", "1.11.*", "1.12.*", "2.0.*" })
+public class MetadataTermMappingResource extends MetadataDelegatingCrudResource<MetadataTermMapping> {
+	
+	public static final String PARAM_SOURCE_NAME = "sourceName";
+	
+	public static final String PARAM_SOURCE_UUID = "sourceUuid";
+	
+	public static final String PARAM_REFERENCE_CLASS = "refClass";
+	
+	public static final String PARAM_REFERENCE_UUID = "refUuid";
+	
+	@Override
+	public Object retrieve(String uniqueIdentifier, RequestContext context) throws ResponseException {
```

Comment:
Typically we do not override the retrieve method. For queries like /metadatatermmapping?sourceUuid=... we implement a SearchHandler see for example https://github.com/openmrs/openmrs-module-webservices.rest/blob/master/omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_8/LocationSearchHandler.java